yacl/crypto/primitives/tpre/tpre.h

```diff
@@ -0,0 +1,126 @@
+ *    plaintext.
+ *
+ * For the description of capsule, you can see in the Capsule class.
+ */
+class TPRE {
+ public:
+  TPRE(){};
+  ~TPRE(){};
+
+  //   /// @brief Rename the re-encryption ciphertext set,  which contains
+  //   capsule
+  //   /// fragments set(CFrags set) and enc_data
+  //   typedef std::pair<std::vector<std::unique_ptr<Capsule::CFrag>>,
+  //                     std::vector<uint8_t>>
+  //       std::pair<std::vector<std::unique_ptr<Capsule::CFrag>>,
+  //                     std::vector<uint8_t>>;
+
+  /// @brief TPRE's encryption algorithm
+  /// @param ecc_group, Algebraic operations are on elliptic curves
+  /// @param pk_A, the public key of Alice, Alice refers to the authorizer
+  /// @param iv, random value
+  /// @param plaintext
+  /// @return capsule and encrypted data
+  std::pair<Capsule::CapsuleStruct, std::vector<uint8_t>> Encrypt(
+      std::unique_ptr<EcGroup> ecc_group, std::unique_ptr<Keys::PublicKey> pk_A,
+      const std::string& iv, const std::string& plaintext);
+
+  /// @brief TPRE's decryption algorithm
+  /// @param ecc_group, Algebraic operations are on elliptic curves
+  /// @param capsule_struct, reference Capsule class.
+  /// @param iv, random value
+  /// @param enc_data, decrypted data
+  /// @param sk_A, the secret key of Alice, Alice refers to the authorizer
+  /// @return plaintext
+  std::string Decrypt(
+      const std::unique_ptr<EcGroup>& ecc_group,
+      const std::unique_ptr<Capsule::CapsuleStruct>& capsule_struct,
+      const std::string& iv, const std::vector<uint8_t>& enc_data,
+      const std::unique_ptr<Keys::PrivateKey>& sk_A) const;
+
+  /// @brief TPRE's threshold proxy re-encryption algorithm
+  /// @param ecc_group, Algebraic operations are on elliptic curves
+  /// @param kfrag, the re-encryption key of proxy
+  /// @param ciphertext, capsule and encrypted data
+  /// @return ciphertext fragments
+  std::pair<Capsule::CFrag, std::vector<uint8_t>> ReEncrypt(
+      const std::unique_ptr<EcGroup>& ecc_group,
+      const std::unique_ptr<Keys::KFrag>& kfrag,
+      std::pair<const std::unique_ptr<Capsule::CapsuleStruct>&,
+                const std::vector<uint8_t>&>
+          ciphertext) const;
+
+  /// @brief TPRE's threshold decryption algorithm
+  /// @param ecc_group, Algebraic operations are on elliptic curves
+  /// @param sk_B, the secret key of Bob
+  /// @param pk_A, the public key of Alice
+  /// @param pk_B, the public key of Bob
+  /// @param iv, random value
+  /// @param C_prime_set, ciphertext fragments set
+  /// @return plaintext
+  std::string DecryptFrags(
+      const std::unique_ptr<EcGroup>& ecc_group,
+      const std::unique_ptr<Keys::PrivateKey>& sk_B,
+      const std::unique_ptr<Keys::PublicKey>& pk_A,
+      const std::unique_ptr<Keys::PublicKey>& pk_B, const std::string& iv,
+      std::pair<const std::vector<std::unique_ptr<Capsule::CFrag>>&,
+                const std::vector<uint8_t>&>
```

Comment:
Suggest be `const std::pair<std::vector<Capsule::CFrag>, std::vector<uint8_t>> & C_prime_set`